src/Common/Auth/OpenIDConnect/Entities/UserEntity.php

```diff
@@ -94,7 +94,7 @@ public function setIdentifier($id): void
 
     protected function getAccountByPassword($userrole, $username, $password, $email = ''): bool
     {
-        if (($userrole === "users") && (($GLOBALS['oauth_password_grant'] === 1) || ($GLOBALS['oauth_password_grant'] === 3))) {
+        if (($userrole == "users") && (($GLOBALS['oauth_password_grant'] == 1) || ($GLOBALS['oauth_password_grant'] == 3))) {
```

Comment:
I wonder if this is best way for handling type differences. i.e Global string to int.
For type safety, should type cast to compared type, however, this may be common practice in PHP, I don't know because i'm still stuff in my years working in strongly typed languages.
You've been doing PHP much longer than I Brady so, are you comfortable with this?

Reply:
Prob can't do strict type comparisons unless type is guaranteed (ie. we create it or it comes from a php function). When using request variables or stuff from database then prob can't do strict type comparisons. Would avoid forcing type also since this just seems to then potentially throw errors in 8 (if the typing is not consistent); ie. damned if we do and slightly less damned if we don't :)

Reply:
gotcha, thanks and makes sense.


> (note this is why null is very nice for default setting of variable since it will avoid typing errors in php8)

still say need to be careful with null concerning database writes